### PR TITLE
WEA-4 Add Navigation Drawer

### PR DESCRIPTION
**Pull request by ayush-shta**
also added baseactivity

### CODE REVIEW

File: app/build.gradle [hunk 1]
```diff
@@ -28,7 +28,11 @@ repositories {
 
 dependencies {
   implementation fileTree(dir: 'libs', include: ['*.jar'])
-  implementation 'com.android.support:appcompat-v7:28.0.0-rc01'
+
+  //downgraded version for design support library
```
Comment (subash-poudel) [inline app/build.gradle:32]: Use the beta version from here: https://developer.android.com/topic/libraries/support-library/revisions

File: app/src/main/java/com/ayush/weatherapp/main/MainActivity.java [hunk 2]
```diff
@@ -1,15 +1,80 @@
...
+    }
+    return super.onOptionsItemSelected(item);
+  }
+
+  private void setActionBar() {
+    setSupportActionBar(toolbar);
+
+    ActionBar actionbar = getSupportActionBar();
+
+    if (actionbar != null) {
```
Comment (subash-poudel) [inline app/src/main/java/com/ayush/weatherapp/main/MainActivity.java]: the null check is not required for action bars it's either there not not there

Comment (ayush-shta) [reply app/src/main/java/com/ayush/weatherapp/main/MainActivity.java]: :ok:

File: app/build.gradle (hunk 1 above)
Comment (ayush-shta) [reply app/build.gradle:32]: :ok:

**Pull request merged**